index.js

```diff
@@ -96,8 +96,7 @@ FibonacciHeap.prototype.extractMinimum = function () {
 
     // Merge the children of the minimum node with the root list
     this.minNode = mergeLists(nextInRootList, extractedMin.child, this.compare);
-    if (nextInRootList) {
-      this.minNode = nextInRootList;
+    if (this.minNode) {
```

Comment:
I like it :+1: thanks for explaining it. I created https://github.com/gwtw/js-fibonacci-heap/issues/10 to follow up with a test case.